readmore.md

```diff
@@ -0,0 +1,186 @@
+# MuSig2 Implementation with libsecp256k1
+
+This is a MuSig2 implementation using the libsecp256k1 library for EC operations.
+
+To install the library, follow the directions in [libsecp256k1](https://github.com/bitcoin-core/secp256k1).
+The library provides an optimized C library for ECDSA signatures and secret and public key operations on curve ecp256k1, as well as usage examples including ECDSA and Schnorr signatures.
+
+This implementation requires configuring the libsecp256k1 with an additional flag `--enable-module-schnorrsig` as stated in [libsecp256k1](https://github.com/bitcoin-core/secp256k1).
+
+Run the example with examplemusig2.c
+
+```
+make ctest
+./ctest_run/ctest
+```
+
+Run the Google tests with
+
+```
+make gtest
+./gtest_run/gtest
+```
+
+## Schnorr Signature of libsecp256k1
+
+The library offers an optional module implementing Schnorr signatures over the curve `secp256k1`.
+The scheme is compatible with [BIP-340](https://github.com/bitcoin/bips/blob/master/bip-0340.mediawiki) that produces 64-byte Schnorr signatures.
+BIP-340 standardized the scheme with some modifications for practical purposes.
+
+### Schnorr Signature in a nutshell
+
+Let $p$ be a prime and $E$ be an elliptic curve defined over the finite field $F_p$ with the base point $G$.
+The signature scheme works as follows:
+
+1. **Key generation:** The secret key $x$ is randomly chosen in $\bmod p$, the  corresponding public key is $X = G \cdot x$.
+
+2. **Signing:** Signer selects a random number $r$ in $\bmod p$, and computes the  nonce $R = G \cdot r$.
+   For the message $m$, signer sets the challenge as $c = H (X, R,  m)$ and computes $s = r + cx$.
+   The signature is obtained as $(R, s)$.
+
+3. **Verification:** The signature is valid if $G \cdot s = R + X \cdot c$.
+
+### Modifications
+
+- #### Encoding nonce and public key:
+Instead of encoding full `X`, `Y` coordinates of $R$ and $X$ (64-byte public key, 96-byte signature), or compressed encoding (33-byte public key, 65-byte signature), BIP-430 preferred to use xonly encoding (32-byte public key, 64-byte signature).
```

Comment:
This seems ambiguous. I'd suggest specifying as follows: 
```suggestion
Instead of encoding full `X`, `Y` coordinates of $X$ and $R$ (which results in a 64-byte public key and 96-byte signature), or compressed encoding (33-byte public key, 65-byte signature respectively), BIP-430 preferred to use xonly encoding (32-byte public key, 64-byte signature).
```

Reply:
You could say a bit more of what xonly is. And also, [this](https://github.com/input-output-hk/musig2/pull/22/files#r1006697187) comment applies here